### ISSUE TITLE
[apex] Apex unit test classes should have at least one System.assert() or assertEquals() or AssertNotEquals() call

### ISSUE THREAD

**Issue by s4SHIVam7** [pmd/pmd#4502]
**Affects PMD Version: 7.0.0** 

**Rule:**

Apex unit test classes should have at least one System.assert() or assertEquals() or AssertNotEquals() call:
<https://docs.pmd-code.org/latest/pmd_rules_apex_bestpractices.html#apexunittestclassshouldhaveasserts>

**Description:**
 In the latest update of salesforce , now we can use Assert.areEqual() instead of system.assertEquals(), but PMD is showing as error.

**Code Sample demonstrating the issue:**

```
String sub = 'abcde'.substring(2);
Assert.areEqual('cde', sub);
```

**Expected outcome:** 

PMD reports a violation at line 2, but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other] [VScode | Cmd]*

**Comment by jsotuyod** [comment]
@s4SHIVam7 thanks for the report.

I fear I can't reproduce this, the methods are considered in the rule:

https://github.com/pmd/pmd/blob/920386d4bd0d521b715769a223578b0e01a751e7/pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/bestpractices/ApexUnitTestClassShouldHaveAssertsRule.java#L40-L41

and there is even a unit test ensuring it works as expected:

https://github.com/pmd/pmd/blob/920386d4bd0d521b715769a223578b0e01a751e7/pmd-apex/src/test/resources/net/sourceforge/pmd/lang/apex/rule/bestpractices/xml/ApexUnitTestClassShouldHaveAsserts.xml#L121-L135

Can you provide more detail as to how you are running PMD?

**Comment by jsotuyod** [comment]
For what I see, that extension allows to update PMD locally https://github.com/ChuckJonas/vscode-apex-pmd#upgrading-pmd

Otherwise, you may want to look at an alternative plugin, but I fear from our end there is little we can do. PMD works as expected.

**Comment by renatoliveira** [comment]
> For what I see, that extension allows to update PMD locally https://github.com/ChuckJonas/vscode-apex-pmd#upgrading-pmd
> 
> Otherwise, you may want to look at an alternative plugin, but I fear from our end there is little we can do. PMD works as expected.

I'm pretty sure it is for [development purposes only](https://github.com/ChuckJonas/vscode-apex-pmd/blob/95b685cce9eddb690834bb812d70e0b126d95406/package.json#L192).

Hopefully we can make some noise in that repo for an update, @s4SHIVam7. ;)

**Comment by s4SHIVam7** [comment]
@jsotuyod As an individual engaged in utilizing PMD analysis for code evaluation, I am presently executing PMD analysis through the Visual Studio Code editor. Specifically, I have installed the Apex PMD extension version 0.5.9 for this purpose. Despite the fact that I have installed the most up-to-date version, I am encountering an error during execution.

**Comment by s4SHIVam7** [comment]
@renatoliveira: We are exactly on the same page, and we will definitely make noise in that repository for an update.
